### DISCUSSION TITLE
Domain events fired before saved to database?

### DISCUSSION THREAD

**Discussion by ppslde** [jasontaylordev/CleanArchitecture#971] [Q&A] [answered]
Hi,

as I understood, a domain event is something which already happend, e.g. UserCreatedEvent.
When I follow the template by adding an Interceptor, which triggers all attached domain events then I cannot get the newly generated id for a entity.

So, in my example I want to create a "UserProfile" entity after a "User" entity was created with added domain event "UserCreatedEvent", but when saving the "UserProfile" entity the id of the "User" entity is still empty as the events are triggered before the save into the database.

Do I miss something or should the domain events triggered after save to database? Currently they are triggered before saving an entity to database.

**Comment by neozhu** [comment, marked as answer]
Hi @ppslde,

I understand your concern. I've adjusted the process: for `DeleteEvent`, the `DispatchDomainEvents` first; for other events, it's done after saving to the database. Hope this helps.
[src/Infrastructure/Persistence/Interceptors/DispatchDomainEventsInterceptor.cs](https://github.com/neozhu/CleanArchitectureWithBlazorServer/blob/main/src/Infrastructure/Persistence/Interceptors/DispatchDomainEventsInterceptor.cs)

**Reply by ppslde** [reply]
Thanks, @neozhu. This approach was my first thought too, so it looks good to me 👍🏼.
